diesel/src/query_builder/update_statement/mod.rs

```diff
@@ -51,6 +51,11 @@ impl<T, U, V, DB> QueryFragment<DB> for UpdateStatement<T, U, V> where
     V: changeset::Changeset<DB>,
 {
     fn to_sql(&self, out: &mut DB::QueryBuilder) -> BuildQueryResult {
+        if self.values.is_noop() {
+            return Err("There are no changes to save. This \
+                       query cannot be built".into())
```

Comment:
As this should be a cold path anyway (is `#[cold]` a thing?), you could create a more fancy error message like ```format!("There are no changes to be written to `{table}`. This query cannot be built.", table=self.table.from_clause())```